src/plugins/score-plugin-packagemanager/PackageManager/View.hpp

```diff
@@ -34,27 +38,40 @@ class PluginSettingsView : public score::GlobalSettingsView
   void handleAddonList(const QJsonObject&);
   void handleAddon(const QJsonObject&);
 
-  void installAddon(const RemotePackage& addon);
-  void installLibrary(const RemotePackage& addon);
-  void installSDK(const RemotePackage& addon);
+  PackagesModel* get_current_model();
+  int get_current_row(const QTableView* t);
+  Package selected_Package(const PackagesModel* model, int row);
 
+  void installAddon(const Package& addon);
+  void installLibrary(const Package& addon);
+  void installSDK(const Package& addon);
+
+  void open_link();
   void install();
+  void uninstall();
   void on_message(QNetworkReply* rep);
 
-  void on_packageInstallSuccess(const RemotePackage& addon, const QDir& destination, const std::vector<QString>& res);
-  void on_packageInstallFailure(const RemotePackage& addon);
+  void on_packageInstallSuccess(const Package& addon, const QDir& destination, const std::vector<QString>& res);
+  void on_packageInstallFailure(const Package& addon);
 
-  QTabWidget* m_widget = new QTabWidget;
+  void set_info();
+  void reset_progress();
+  void progress_from_bytes(qint64 bytesRecieved, qint64 bytesTotal);
 
-  QTableView* m_addonsOnSystem{new QTableView};
+  QWidget* m_widget{new QWidget};
 
+  QTableView* m_addonsOnSystem{new QTableView};
   QTableView* m_remoteAddons{new QTableView};
-  QPushButton* m_refresh{new QPushButton{tr("Refresh")}};
+
+  QPushButton* m_link{new QPushButton{tr("Open link")}};
   QPushButton* m_install{new QPushButton{tr("Install")}};
+  QPushButton* m_uninstall{new QPushButton{tr("Uninstall")}};
 
   QProgressBar* m_progress{new QProgressBar};
   QNetworkAccessManager mgr;
-
   int m_addonsToRetrieve = 0;
+
+  QStorageInfo storage{score::AppContext().settings<Library::Settings::Model>().getPackagesPath()};
```

Comment:
maybe initialize that one in the .cpp instead in order to not leak the needed headers for that in the .hpp

Reply:
(mainly for reducing compile times)